docs/adr/0008_signing_verification.md

```diff
@@ -0,0 +1,358 @@
+
+Some workflows require an additional **OCI Cosign** signature over the same canonical payload, optionally uploaded to an OCI registry and recorded in a transparency log (Rekor).
+
+---
+
+## Decision
+
+### Two-step flow
+
+1. **Digest calculation** mutates the descriptor to embed required digests (resources, component references, component-version).
+2. **Signing** appends a signature envelope without computing or embedding digests. Optionally, the same payload may be **cosigned**.
+
+### Commands (verb–noun, long options only, no stdin)
+
+```bash
+# Step 1 — Digest calculation (mutates descriptor: digests only)
+ocm add digests cv <ref> --recurse --force
+
+# Step 2 — Signing (no digest mutation)
+# Direct signing via a single --cert (+ optional password).
+# Optional OCI Cosign signing for the same canonical payload.
+ocm sign cv <ref> \
+  --sig <slot> \
+  --pin <sha256:...> \
+  --cert <path> \
+  --password <pw> \
+  --password-prompt \
+  --cosign \
+  --cosign-upload \
+  --cosign-skip-tlog \
+  --cosign-annotation <key=value> \
+  --cosign-annotation <key=value> \
+  --cosign-identity-token <token-or-path>
+```
+
+- `<ref>`: descriptor reference (file path or repository reference).
+- `--recurse`: calculate digests for referenced component versions.
+- `--force`: overwrite existing digest fields.
+- `--sig <slot>`: logical signature slot/name (e.g., `mysig@1234`).
+- `--pin <sha256:...>`: expected component-version digest; fail on mismatch.
+- `--cert <path>`: certificate/key material for direct signing (PEM/PKCS#8/PKCS#12).
+- `--password <pw>` / `--password-prompt`: decrypt key material if protected.
+- `--cosign`: also produce an OCI Cosign signature for the same canonical payload.
+- `--cosign-upload`: if `<ref>` resolves to an OCI subject, attach/upload the Cosign signature to the registry; otherwise, store the Cosign bundle only.
+- `--cosign-skip-tlog`: disable Rekor upload.
+- `--cosign-annotation`: repeatable `key=value` annotations for the Cosign signature.
+- `--cosign-identity-token <token-or-path>`: pass an OIDC ID token directly (alternatively, use `SIGSTORE_ID_TOKEN` env).
+
+### Normalization
+
+- Canonicalization for component-version digest and signature payload is **JCS (RFC 8785)** via the existing `jcs` package.
+- No CLI flag to select a different normalization.
+
+---
+
+## Behavior
+
+### Digest calculation (`ocm add digests cv`)
+
+- Computes and **embeds** digests:
+  - **Resources**: content digests.
+  - **Component references**: digest of the referenced component descriptor.
+  - **Component-version**: digest over the canonicalized descriptor.
+- May download artifacts and referenced descriptors as needed.
+- Writes the updated descriptor back to `<ref>`.
```

Comment:
So its not just a digest calculation, but rather a complete update to the component version right?